### PR TITLE
feat: Tree View in List

### PR DESCRIPTION
**Pull request by hrwX**
- Incorporate TreeView in Frappe ListView
- image

- **Why a new view for Tree when we already have TreeView ?**
        - The original TreeView, take the customer away from the ListView.
        - Since the Treeview is a custom implementation, the user cannot leverage the filtering option of ListView
- **How does it work?**
        - If a doctype has a tree view enabled, we will be able to see the new tree list view
- **what is considered a child on the tree? child tables? custom?**
        - Lets take an example of Task
            - We have Task A and Task B, and Task B is a task nested under Task A
            - Now in Task DocType, when we add parent_task in Task B as Task A, Task B becomes a child of Task A, and that is when the nested view will be visible for Task A
- **is this for specific doctypes? is it available for all doctypes and how does it behave?**
        - This is just available for doctypes with tree view enabled in doctypes, ie doctypes like accounts, tasks.

### CODE REVIEW

**Comment by DeveloperMetal** [comment]
@hrwX can you add some info on the feature?

- how does it work?
- what is considered a child on the tree? child tables? custom?
- is this for specific doctypes? is it available for all doctypes and how does it behave?

**Comment by hrwX** [comment]
@DeveloperMetal

Why a new view for Tree when we already have TreeView ?
- The original TreeView, take the customer away from the ListView.
- Since the Treeview is a custom implementation, the user cannot leverage the filtering option of ListView

How does it work?
- If a doctype has a tree view enabled, we will be able to see the new tree list view

what is considered a child on the tree? child tables? custom?
- Lets take an example of Task
- We have Task A and Task B, and Task B is a task nested under Task A
- Now in Task DocType, when we add parent_task in Task B as Task A, Task B becomes a child of Task A, and that is when the nested view will be visible for Task A

is this for specific doctypes? is it available for all doctypes and how does it behave?
- This is just available for doctypes with tree view enabled in doctypes, ie doctypes like accounts, tasks.

File: frappe/public/js/frappe/views/tree/tree_view.js [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+
+		let list = $row.find(`.list-nested-row-container`);
+		let level = parseInt($row[0].getAttribute("data-level")) + 1;
+
+		let $list = $(list);
+
+		$row.toggleClass("opened");
+		$list.toggleClass("hide");
+
+		console.log($list);
```
Comment (neilLasrado) [inline frappe/public/js/frappe/views/tree/tree_view.js]: remove this console.log

**Review by neilLasrado — DISMISSED** [review]
except for the one console.log, rest looks good

**Comment by neilLasrado** [comment]
@hrwX what else is pending in this? Let's get this deployed on bloomstack.garden or a similar site for testing

**Comment by neilLasrado** [comment]
To be deployed on a test server for testing

**Comment by sahil28297** [comment]
Comment to make it active again. whoosh

**Comment by neilLasrado** [comment]
@nehasacher143 @hrwX Is this PR ready for review now?

**Comment by hrwX** [comment]
@neilLasrado @sahil28297 PRs ready to go!!!

**Review by neilLasrado — DISMISSED** [review]

**Comment by neilLasrado** [comment]
@sahil28297 do we merge this to develop?

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**